cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -1007,6 +1019,63 @@ func deleteAllPodsBoundToNode(ctx *controllerContext, nodeName string) error {
 	return utilerrors.NewAggregate(errs)
 }
 
+func markFailedAndDeletePodWithCondition(ctx *controllerContext, pod *v1.Pod, condition *corev1apply.PodConditionApplyConfiguration) error {
+	fieldManager := "GCPControllerManager"
+
+	// Copied directly from pod gc controller (https://github.com/kubernetes/kubernetes/blob/17bf864c1fc20e4badc9a6e0659c5190310462e1/pkg/controller/podgc/gc_controller.go)
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+
+		// Extact the pod status as GCPControllerManager may or may not own the pod phase, if
+		// it owns the phase then we need to send the field back if the condition
+		// is added.
+		podApply, err := corev1apply.ExtractPodStatus(pod, fieldManager)
+		if err != nil {
+			return nil
+		}
+
+		// Set the status in case PodGC does not own any status fields yet
+		if podApply.Status == nil {
+			podApply.WithStatus(corev1apply.PodStatus())
+		}
+
+		updated := false
+		if condition != nil {
+			updatePodCondition(podApply.Status, condition)
+			updated = true
+		}
+		// Mark the pod as failed - this is especially important in case the pod
+		// is orphaned, in which case the pod would remain in the Running phase
+		// forever as there is no kubelet running to change the phase.
```

Comment:
Make this comment specific to this controller.

IIUC, the assumption here is that there is no kubelet at all?

Reply:
It is correct, no kubelet will be running to change the phase, updated.